src/Authenticator/CookieAuthenticator.php

```diff
@@ -134,9 +134,14 @@ protected function _createPlainToken(ArrayAccess|array $identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        $salt = $this->getConfig('salt', '');
+        if ($identity[$usernameField] === null || $identity[$passwordField] === null) {
+            throw new InvalidArgumentException(
```

Comment:
Wouldn't this create a 500 for the application? Could we use an exception that will render a 40x response instead?

Reply:
It should, it is a developer error.
It can only run into this if you misconfigured them afaik.
As otherwise it shouldnt even go into this authenticator that far.